src/sage/matroids/graphic_matroid.pyx

```diff
@@ -720,32 +707,32 @@ def _closure(self, X):
             sage: sorted(M._closure([4]))
             [0, 4, 5]
         """
-        X = set(X)
-        Y = self.groundset().difference(X)
-        edgelist = self._groundset_to_edges(Y)
-        g = self._subgraph_from_set(X)
-        V = g.vertices(sort=False)
-        components = g.connected_components_number()
+        cdef set XX = set(X)
+        cdef frozenset Y = self.groundset().difference(XX)
+        cdef list edgelist = self._groundset_to_edges(Y)
+        cdef GenericGraph_pyx g = self._subgraph_from_set(XX)
+        cdef list V = g.vertices(sort=False)
+        cdef int components = g.connected_components_number()
         for e in edgelist:
```

Comment:
```suggestion
        cdef frozenset Y = self._groundset.difference(XX)
        cdef list edgelist = self._groundset_to_edges(Y)
        cdef GenericGraph_pyx g = self._subgraph_from_set(XX)
        cdef list V = g.vertices(sort=False)
        cdef int components = g.connected_components_number()
        cdef tuple e
        for e in edgelist:
```
Since the edge is going to be a tuple and perhaps explicit type casting to avoid unnecessary type checking.